src/v2/BaseLoginRouter.js

```diff
@@ -151,17 +171,21 @@ export default Router.extend({
     // and remove it from `settings` afterwards as IDX response always has
     // state token (which will be set into AppState)
     if (this.settings.get('stateToken')) {
-      return introspectStateToken(this.settings)
-        .then(idxResp => {
-          this.settings.unset('stateToken');
-          this.appState.trigger('remediationSuccess', idxResp);
-          this.render(Controller, options);
-        })
-        .catch(errorResp => {
-          this.settings.unset('stateToken');
-          this.appState.trigger('remediationError', errorResp.error);
-          this.render(Controller, options);
-        });
+      if (this.settings.options.deviceEnrollment && this.settings.options.deviceEnrollment.name) {
```

Comment:
remove this `if`.. and all terminal state with no remediation should be part of else statement.

Reply:
use `terminalStateWithNoRemediation` that passed as options to check for no remediation terminal state..

basically
```
if (this.settings.get('stateToken') && !options.terminalStateWithNoRemediation`) 
--- idx flow
else 
--- terminal states
```